app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  validates_presence_of :name, :email, :google_token
+  
+  def self.from_omniauth(auth)
+    where(email: auth["info"]["email"]).first_or_initialize do |user|
+      user.name = auth["info"]["name"]
+      user.email = auth["info"]["email"]
+    end
```

Comment:
`first_or_initialize` either returns the `user` present in the database with the matching `email` or creates it